### PR TITLE
✨ Support a list of maps in tilt-settings

### LINKED ISSUE #3299: Add support for multiple Tilt providers in tilt-provider.json
**User Story**

As a developer I want to be able to define multiple Tilt providers per repository.
This is useful for developing CAPI providers that include multiple operators (main.go).

**Detailed Description**

Let’s say I am working on a new CAPA operator/controller in `bootstrap/eks`.
My `tilt-provider.json` might look like this:

```json
cat tilt-provider.json
{
  "items": [
    {
      "name": "aws",
      "config": {
        "context": "",
        "image": "gcr.io/k8s-staging-cluster-api-aws/cluster-api-aws-controller",
        "live_reload_deps": [
          "main.go",
          "go.mod",
          "go.sum",
          "api",
          "cmd",
          "controllers",
          "pkg"
        ]
      }
    },
    {
      "name": "eks-bootstrap",
      "config": {
        "context": "bootstrap/eks",
        "image": "gcr.io/k8s-staging-cluster-api-aws/eks-bootstrap-controller",
        "live_reload_deps": [
          "main.go",
          "go.mod",
          "go.sum",
          "api",
          "controllers"
        ]
      }
    }
  ]
}
```

**Anything else you would like to add:**

FWIW I have this hack right now but it has two bugs, see the TODO comments:

```patch
git diff
diff --git a/Tiltfile b/Tiltfile
index 89a8f23cf..82bedbff0 100644
--- a/Tiltfile
+++ b/Tiltfile
@@ -103,10 +103,11 @@ def load_provider_tiltfiles():
     for repo in provider_repos:
         file = repo + "/tilt-provider.json"
         provider_details = read_json(file, default = {})
-        provider_name = provider_details["name"]
-        provider_config = provider_details["config"]
-        provider_config["context"] = repo
-        providers[provider_name] = provider_config
+        for item in provider_details["items"]: # TODO: Make it backward compatible.
+            provider_name = item["name"]
+            provider_config = item["config"]
+            provider_config["context"] = repo + "/" + provider_config["context"] # TODO: context might not be defined
+            providers[provider_name] = provider_config

 tilt_helper_dockerfile_header = """
 # Tilt image
```

/kind feature

### PR DESCRIPTION
**Pull request by h0tbird**
**What this PR does / why we need it**:
As a developer I want to be able to define multiple Tilt providers per repository.
This is useful for developing CAPI providers that include multiple operators.

**Anything else you would like to add:**
My CAPI `tilt-settings.json` looks like this:
```
{
  "default_registry": "localhost:5000",
  "provider_repos": ["../cluster-api-provider-aws"],
  "enable_providers": ["aws", "eks-bootstrap"],
  "kustomize_substitutions": {
    "AWS_B64ENCODED_CREDENTIALS": "<REDACTED>",
    "EXP_MACHINE_POOL:=false": "false",
    "EXP_CLUSTER_RESOURCE_SET:=false": "false"
  }
}
```
And I tested two different CAPA `tilt-settings.json`:
```
{
  "name": "aws",
  "config": {
    "image": "gcr.io/k8s-staging-cluster-api-aws/cluster-api-aws-controller",
    "live_reload_deps": [
      "main.go", "go.mod", "go.sum", "api", "cmd", "controllers", "pkg"
    ]
  }
}
```
and this:
```
[
  {
    "name": "aws",
    "config": {
      "image": "gcr.io/k8s-staging-cluster-api-aws/cluster-api-aws-controller",
      "live_reload_deps": [
        "main.go", "go.mod", "go.sum", "api", "cmd", "controllers", "pkg"
      ]
    }
  },
  {
    "name": "eks-bootstrap",
    "config": {
      "context": "bootstrap/eks",
      "image": "gcr.io/k8s-staging-cluster-api-aws/eks-bootstrap-controller",
      "live_reload_deps": [
        "main.go", "api", "controllers"
      ]
    }
  }
]
```
**Which issue(s) this PR fixes**:
Fixes #3299

### CODE REVIEW

File: Tiltfile [hunk 1]
```diff
@@ -103,10 +105,16 @@ def load_provider_tiltfiles():
     for repo in provider_repos:
         file = repo + "/tilt-provider.json"
         provider_details = read_json(file, default = {})
-        provider_name = provider_details["name"]
-        provider_config = provider_details["config"]
-        provider_config["context"] = repo
-        providers[provider_name] = provider_config
+        if "name" in provider_details:
```
Comment (fabriziopandini) [inline Tiltfile:108]: what about using `if type(provider_details) != type([])` (so we are a little bit more explicit about the intent)?

Comment (h0tbird) [reply Tiltfile:108]: Much better, thanks @fabriziopandini

**Comment by fabriziopandini** [comment]
/lgtm
please squash commits

**Comment by fabriziopandini** [comment]
/lgtm
over to @ncdc or @detiber for the final approval

**Pull request merged**